### PR TITLE
Fix(validation)  add trading params validation

### PR DESCRIPTION
**Pull request by Aciila**
## Summary by CodeRabbit

- **New Features**
  - Added comprehensive client-side validation for swap and limit orders, with clearer human-friendly error reporting and runtime validation when posting orders (includes SDK-level pre-checks and logging).
  - Exposed validation helpers in the public API for use by client apps.

- **Tests**
  - Added extensive validation unit tests covering many edge cases and chains.
  - Updated test fixtures to use realistic token and receiver addresses.

### CODE REVIEW

**Comment by Aciila** [comment]
I have read the CLA Document and I hereby sign the CLA

File: packages/trading/src/tradingSdk.ts [hunk 1]
```diff
@@ -109,14 +110,18 @@ export class TradingSdk {
     params: WithPartialTraderParams<TradeParameters>,
     advancedSettings?: SwapAdvancedSettings,
   ): Promise<OrderPostingResult> {
-    return postSwapOrder(this.mergeParams(params), advancedSettings, this.options.orderBookApi)
+    const mergedParams = this.mergeParams(params)
+    validateSwapParameters(mergedParams)
```
Comment (shoom3301) [inline packages/trading/src/tradingSdk.ts:114]: `validateSwapParameters` is already called inside of `postSwapOrder`, so we don't need to call it here as well

File: packages/trading/src/validation.ts [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    this.name = 'TradeValidationError'
+  }
+}
+
+/**
+ * Configuration for validation rules
+ */
+interface ValidationConfig {
+  readonly maxSlippageBps: number
+  readonly minAmount: string
```
Comment (shoom3301) [inline packages/trading/src/validation.ts:29]: We prefer using native `bigint` instead of string

File: packages/trading/src/validation.ts [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+  readonly maxValidityPeriod: number
+}
+
+/**
+ * Default validation configuration
+ */
+const DEFAULT_CONFIG: ValidationConfig = {
+  maxSlippageBps: 5000,
+  minAmount: '1',
+  maxValidityPeriod: 86400,
```
Comment (shoom3301) [inline packages/trading/src/validation.ts:39]: There is a const `MAX_VALID_TO_EPOCH` in config package

File: packages/trading/src/validation.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+ * Validates that validity period is reasonable
+ */
+function isValidValidityPeriod(validFor: number): boolean {
+  return validFor > 0 && validFor <= DEFAULT_CONFIG.maxValidityPeriod
+}
+
+/**
+ * Validates base trade common fields
+ */
+function validateCommonFields(params: {
```
Comment (shoom3301) [inline packages/trading/src/validation.ts:95]: `params` should have `SwapParameters` type

File: packages/trading/src/validation.ts [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+  const summary = errors.map((error, index) => `${index + 1}. ${error.field}: ${error.message}`).join('\n')
+
+  return `Found ${errors.length} validation error${errors.length > 1 ? 's' : ''}:\n${summary}`
+}
+
+/**
+ * Validates common edge cases that can cause silent failures
+ */
+export function validateTradeAmounts(sellAmount: string, buyAmount: string, slippageBps?: number): void {
```
Comment (shoom3301) [inline packages/trading/src/validation.ts:265]: I don't think `validateTradeAmounts` is correct, we acrually allow amounts less than 1000 wei

File: packages/trading/src/validation.ts [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+
+  if (buyBn < BigInt('1000')) {
+    throw new TradeValidationError(
+      'Buy amount is too small (dust). Minimum recommended: 1000 wei',
+      'buyAmount',
+      buyAmount,
+    )
+  }
+
+  if (slippageBps && slippageBps > 1000) {
```
Comment (shoom3301) [inline packages/trading/src/validation.ts:285]: This is also not correct, slippage might be even 100%
